### PR TITLE
[dbnode] Add placement set handler for perform set operations

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Allows us to do goal state based placement updates, must be used with care!!

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

**Comment by robskillington** [comment]
TODO tests

File: src/query/api/v1/handler/placement/set.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+package placement
+
+import (
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/m3db/m3/src/cluster/placement"
+
```
Comment (schallert) [inline src/query/api/v1/handler/placement/set.go]: Nit: extra space?

File: src/query/api/v1/handler/placement/set.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	if err != nil {
+		logger.Error("unable to create placement service", zap.Error(err))
+		xhttp.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	curPlacement, err := service.Placement()
+	if err != nil {
+		logger.Error("unable to get current placement", zap.Error(err))
+		xhttp.Error(w, err, http.StatusInternalServerError)
```
Comment (schallert) [inline src/query/api/v1/handler/placement/set.go]: Let's make this `StatusNotFound`? Thinking from the operator perspective, there are times where we have to type check a 404 from Coordinator API.

Comment (robskillington) [reply src/query/api/v1/handler/placement/set.go]: Sounds good, I added this.

**Pull request merged**